### PR TITLE
OnlineStore 17

### PR DESCRIPTION
**Pull request by Antilevskaya-Ksenia-153501**
Task about ExceptionMiddleware

### CODE REVIEW

File: OnlineStore/OnlineStore.Api/Middlewares/GlobalExceptionHandlingMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    private async Task HandleExceptionAsync(HttpContext context, Exception ex)
+    {
+        var logMessage = $"---> Request {context.Request.Path} failed with error: {ex.Message}";
+        _logger.Information(logMessage);
+
+        context.Response.ContentType = "application/json";
+        context.Response.StatusCode = StatusCodes.Status500InternalServerError;
+
+        ErrorResponse errorResponse = new()
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Api/Middlewares/GlobalExceptionHandlingMiddleware.cs]: use problem details class

File: OnlineStore/OnlineStore.Api/Program.cs [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 builder.Services.AddControllers();
 
 builder.Services.AddApplicationLayer();
 builder.Services.AddDataAccessLayer(builder.Configuration);
 
+var configuration = new ConfigurationBuilder()
+        .SetBasePath(Directory.GetCurrentDirectory())
+        .AddJsonFile("appsettings.json")
+        .AddJsonFile($"appsettings.{Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Production"}.json", true)
+        .Build();
```
Comment (SUPERKISA8BIIT) [inline OnlineStore/OnlineStore.Api/Program.cs:18]: this is unnecessary

**Review by SUPERKISA8BIIT — CHANGES_REQUESTED** [review]
checked

File: OnlineStore/OnlineStore.Api/Middlewares/GlobalExceptionHandlingMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        catch (Exception ex)
+        {
+            await HandleExceptionAsync(context, ex);
+        }
+    }
+
+    private async Task HandleExceptionAsync(HttpContext context, Exception ex)
+    {
+        var logMessage = $"---> Request {context.Request.Path} failed with error: {ex.Message}";
+        _logger.Information(logMessage);
```
Comment (SergeyBrykulskii) [inline OnlineStore/OnlineStore.Api/Middlewares/GlobalExceptionHandlingMiddleware.cs]: .Error

**Pull request merged**